pkg/util/informermanager/informermanager.go

```diff
@@ -0,0 +1,287 @@
+
+	queue  workqueue.Interface
+	logger klog.Logger
+}
+
+func NewInformerManager(client dynamic.Interface, ftcInformer fedcorev1a1informers.FederatedTypeConfigInformer) InformerManager {
+	manager := &informerManager{
+		lock:                      sync.RWMutex{},
+		started:                   false,
+		client:                    client,
+		ftcInformer:               ftcInformer,
+		eventHandlerGenerators:    []*EventHandlerGenerator{},
+		gvrMapping:                tools.NewBijectionMap[string, schema.GroupVersionResource](),
+		informers:                 map[string]informers.GenericInformer{},
+		informerStopChs:           map[string]chan struct{}{},
+		eventHandlerRegistrations: map[string]map[*EventHandlerGenerator]cache.ResourceEventHandlerRegistration{},
+		lastAppliedFTCsCache:      map[string]map[*EventHandlerGenerator]*fedcorev1a1.FederatedTypeConfig{},
+		queue:                     workqueue.NewRateLimitingQueue(workqueue.DefaultItemBasedRateLimiter()),
+		logger:                    klog.LoggerWithName(klog.Background(), "informer-manager"),
+	}
+
+	ftcInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { manager.enqueue(obj) },
+		UpdateFunc: func(_ interface{}, obj interface{}) { manager.enqueue(obj) },
+		DeleteFunc: func(obj interface{}) { manager.enqueue(obj) },
+	})
+
+	return manager
+}
+
+func (m *informerManager) enqueue(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+	if err != nil {
+		m.logger.Error(err, "Failed to enqueue FederatedTypeConfig")
+		return
+	}
+	m.queue.Add(key)
+}
+
+func (m *informerManager) worker() {
+	key, shutdown := m.queue.Get()
+	if shutdown {
+		return
+	}
+	defer m.queue.Done(key)
+
+	_, name, err := cache.SplitMetaNamespaceKey(key.(string))
+	if err != nil {
+		m.logger.Error(err, "Failed to process FederatedTypeConfig")
+		return
+	}
+
+	ftc, err := m.ftcInformer.Lister().Get(name)
+	if apierrors.IsNotFound(err) {
+		if err := m.processFTCDeletion(name); err != nil {
+			m.logger.Error(err, "Failed to process FederatedTypeConfig, will retry")
+			m.queue.Add(key)
+			return
+		}
+		return
+	}
+	if err != nil {
+		m.logger.Error(err, "Failed to process FederatedTypeConfig, will retry")
+		m.queue.Add(key)
+		return
+	}
+
+	err, needReenqueue := m.processFTC(ftc)
+	if err != nil {
+		if needReenqueue {
+			m.logger.Error(err, "Failed to process FederatedTypeConfig, will retry")
+		} else {
+			m.logger.Error(err, "Failed to process FederatedTypeConfig")
+		}
+	}
+	if needReenqueue {
+		m.queue.Add(key)
+	}
+}
+
+func (m *informerManager) processFTC(ftc *fedcorev1a1.FederatedTypeConfig) (err error, needReenqueue bool) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	ftcName := ftc.Name
+	apiResource := ftc.GetSourceType()
+	gvr := schemautil.APIResourceToGVR(&apiResource)
+
+	var informer informers.GenericInformer
+
+	if oldGVR, exists := m.gvrMapping.Lookup(ftcName); exists {
```

Comment:
Why do we need to maintain a one-to-one mapping from ftcName to GVR? Would it be possible to use GVR as the single unique identifier for an informer (even if a GVR was previously managed by a different FTC)?